tests/manual/util/src/com/oracle/util/testing/ManualTestWindow.java

```diff
@@ -0,0 +1,279 @@
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package com.oracle.util.testing;
+
+import java.util.Objects;
+import java.util.function.Supplier;
+import javafx.application.Application;
+import javafx.application.Platform;
+import javafx.geometry.Insets;
+import javafx.scene.Node;
+import javafx.scene.Scene;
+import javafx.scene.control.Button;
+import javafx.scene.control.ContextMenu;
+import javafx.scene.control.MenuItem;
+import javafx.scene.effect.BlurType;
+import javafx.scene.effect.DropShadow;
+import javafx.scene.input.Clipboard;
+import javafx.scene.input.ClipboardContent;
+import javafx.scene.layout.Background;
+import javafx.scene.layout.BorderPane;
+import javafx.scene.layout.HBox;
+import javafx.scene.layout.Priority;
+import javafx.scene.layout.Region;
+import javafx.scene.layout.VBox;
+import javafx.scene.paint.Color;
+import javafx.scene.text.Text;
+import javafx.scene.text.TextFlow;
+import javafx.stage.Stage;
+
+/**
+ * Provides a framework for manual tests to display test instructions, test pane, and
+ * Pass/Fail buttons.
+ * <p>
+ * A simple test would look like this:
+ * <pre>{@code
+ * public class SampleManualTest {
+ *     public static void main(String[] args) throws Exception {
+ *         ManualTestWindow.builder().
+ *             title("Sample Manual Test").
+ *             instructions(
+ *                 """
+ *                 Provide
+ *                 multi-line instructions here.
+ *                 """
+ *             ).
+ *             ui(() -> createTestUI()).
+ *             buildAndRun();
+ *     }
+ *
+ *     private static Node createTestUI() {
+ *         return new Label("Test UI");
+ *     }
+ * }
+ * }</pre>
+ * <p>
+ * The framework will create the test Application, generate the Node representing the test UI,
+ * and show the test Stage, making sure the last two operations happen in the context of the
+ * JavaFX application thread.
+ * <p>
+ * The {@link Builder#ui(Supplier)} method accepts interfaces which create the test Node.
+ * For tests that require multiple windows, these windows can be created by the code specified
+ * with {@link Builder#runAfter(Runnable)}.
+ */
+// TODO timeout
+// TODO log area
+// TODO screenshots on failure?
+// TODO initial position?
```

Comment:
In general I do not like TODO in comments. You can file follow up issue to improve/extend `ManualTestWindow` if needed.